edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/commands/PrepareUpdateCommand.cs

```diff
@@ -1,30 +1,48 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Agent.Edgelet.Commands
 {
+    using System;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.PlanRunner;
     using Microsoft.Azure.Devices.Edge.Agent.Edgelet.Models;
     using Microsoft.Azure.Devices.Edge.Util;
 
     public class PrepareUpdateCommand : ICommand
     {
         readonly IModuleManager moduleManager;
         readonly ModuleSpec module;
+        readonly ModuleUpdateMode moduleUpdateMode;
 
-        public PrepareUpdateCommand(IModuleManager moduleManager, IModule module, object settings)
+        public PrepareUpdateCommand(IModuleManager moduleManager, IModule module, object settings, ModuleUpdateMode moduleUpdateMode)
         {
             this.moduleManager = moduleManager;
             this.module = BuildModuleSpec(module, settings);
+            this.moduleUpdateMode = moduleUpdateMode;
         }
 
         public string Id => $"PrepareUpdateCommand({this.module.Name})";
 
         public string Show() => $"Prepare module {this.module.Name}";
 
-        public Task ExecuteAsync(CancellationToken token)
+        public async Task ExecuteAsync(CancellationToken token)
         {
-            return this.moduleManager.PrepareUpdateAsync(this.module);
+            try
+            {
+                await this.moduleManager.PrepareUpdateAsync(this.module);
+            }
+            catch (Exception e)
```

Comment:
can use: catch (Exception e) when (this.moduleUpdateMode == ModuleUpdateMode.WaitForAll)
 or create a new PrepareUpdateCommand class which throws